libs/core-ui/src/lib/Interfaces/IDataBalanceMeasures.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+export interface IDataBalanceMeasures {
+  AggregateBalanceMeasures?: IAggregateBalanceMeasures;
+  DistributionBalanceMeasures?: IDistributionBalanceMeasures;
+  FeatureBalanceMeasures?: IFeatureBalanceMeasures;
+}
+
+export interface IAggregateBalanceMeasures {
+  AtkinsonIndex: number;
+  TheilLIndex: number;
+  TheilTIndex: number;
+}
+
+export interface IDistributionBalanceMeasures {
+  [key: string]: IDistributionBalanceMeasure;
+}
+
+export interface IDistributionBalanceMeasure {
+  ChiSquarePValue: number;
+  ChiSquareStat: number;
+  InfiniteNormDist: number;
+  JensenShannonDist: number;
+  KLDivergence: number;
```

Comment:
do we really need to compute so many data balance measures for the dataset?  I'm just wondering if this might become too slow in some cases, and if it may be useful to allow users to only compute a subset of these - in which case perhaps instead of this interface something more dynamic like a map should be used from balance measure name to value.  Not meant to block this PR though, just something to think about.  Do any of these measures become much slower for larger data in terms of # of rows or # of features?

Reply:
Map seems like a really good idea. That way, we can add/remove measures in the future with less code changes. Both UX and Python side would be touched, but atleast the interface would not be touched.

Also, I did a small benchmark in the Python side PR so we can better understand the compute time of various dataset sizes, [adding the link here](https://github.com/microsoft/responsible-ai-toolbox/pull/1485#discussion_r914225527).

Reply:
I had an easy time doing this for `IAggregateBalanceMeasures` and `IDistributionBalanceMeasure` since their interfaces only contain str -> number pairs.

For `IFeatureBalanceMeasure`, since each measure has both str -> number and str -> str pairs:
```
{
  ClassA: "Bachelors",
  ClassB: "HS-grad",
  JaccardIndex: 0.1009922102843339,
  KendallRankCorrelation: 1.9020631282523714,
  LogLikelihoodRatio: 0.27920025739587184,
  PointwiseMutualInfo: 0.9519638161627417,
  SorensonDiceCoeff: 0.07642301570298538,
  StatisticalParity: 0.25222421370244086,
  TTest: 0.4236184547392873,
  TTestPValue: 0.342278369038513
},
```
I'm having a hard time defining the interface to where both `ClassA` `ClassB` and the measures can be specified. Is there any way I can do this without changing the schema of what the Python side creates?

![image](https://user-images.githubusercontent.com/64443771/177430003-596710c2-1e24-4cc7-87e0-df2f5e858a06.png)

Reply:
I was able to fix it by allowing the indexable type to also returning a `string`

```
export interface IFeatureBalanceMeasure {
  [key: string]: number | string;
  ClassA: string;
  ClassB: string;
}

```

Reply:
Updated the PR to use indexable type instead of hardcoded measure names .

I'm a huge fan of this change, thank you for the suggestion Ilya! 😊

I'm resolving this thread and we can talk about compute time / slowness [in the other PR](https://github.com/microsoft/responsible-ai-toolbox/pull/1485#discussion_r914225527).